zooniverse/front-end-monorepo
# App-project: refactor when notifications requests called

## Package
- app-project

## Linked Issue and/or Talk Post
- closes #3941 
- closes #3672

## Describe your changes
- add test for anonymous user to not request notifications
- refactor notifications request to `UserPersonalization` load action
- refactor tests per changes

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX?
  - any project, here's production TESS - https://local.zooniverse.org:3000/projects/nora-dot-eisner/planet-hunters-tess?env=production&demo=true
- What user actions should my reviewer step through to review this PR?
  - with notifications (Messages or Notifications in header, let me know if you'd like me to send you a message to create some), start on https://local.zooniverse.org:3000/projects/nora-dot-eisner/planet-hunters-tess?env=production&demo=true
  - go to Talk, will 404, but then click back
  - confirm notification counts still show
- Which storybook stories should be reviewed? N/A
- Are there plans for follow up PR’s to further fix this bug or develop this feature? no, though other notifications related work is ongoing

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Bug Fix
